bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/KSEdgeService.java

```diff
@@ -41,4 +43,14 @@ default KSServiceHandle spot(KSListener ksListener, AudioStream audioStream, Loc
             throws KSException {
         throw new KSException("An edge keyword spotter is not meant to process audio in the server");
     }
+
+    @Override
+    default Set<AudioFormat> getSupportedFormats() {
```

Comment:
I would rather not have default implementations here, as normally I would expect that you need to specify the formats.

Reply:
I missed these when adding this interface on a previous PR.
To give it some context, the KSEdgeService is meant to represent a keyword spotting service which is running outside the server without using the source audio stream. I created it to allow a way of dialog where the server is not constantly streaming the audio from the source to feed the keyword spotting service.
Looking at it now I think I messed up a little, because it seems more correct to create a DialogTriggerService interface and change the KeywordSpotterService interface to implement that one, because at the end you don't what is happening in the remote to trigger the dialog. 
I can change that if you think is a better option, or create a PR to change it afterwards, for now it's not used anywhere else.

Reply:
> Looking at it now I think I messed up a little, because it seems more correct to create a DialogTriggerService interface and change the KeywordSpotterService interface to implement that one, because at the end you don't what is happening in the remote to trigger the dialog.

Yes, I agree.
I am working on an addon for the Wyoming protocol, from the Home Assistant voice ecosystem, and a remote trigger for wakeword is a part of it that I definitely want to support. The architecture you are reffering to is definitely something I was planning to work on.
Also for example, I think that a switch should also be able to trigger a dialog, so a generic DialogTriggerService for those other cases would be a good step forward to support them 👍

Reply:
I'll add a DialogTriggerService interface then and remove the KSEdgeService one.